src/net_processing.cpp

```diff
@@ -1883,7 +1890,9 @@ void PeerManagerImpl::ProcessGetBlockData(CNode& pfrom, Peer& peer, const CInv&
 
 CTransactionRef PeerManagerImpl::FindTxForGetData(const CNode& peer, const GenTxid& gtxid, const std::chrono::seconds mempool_req, const std::chrono::seconds now)
 {
-    auto txinfo = m_mempool.info(gtxid);
+    if (!m_mempool) return {};
```

Comment:
```suggestion
    if (!Assume(m_mempool)) return {};
```

I think debug builds should fail, as it is an oversight to call this function without a mempool.

Reply:
Introduced in this PR, `!m_mempool` implies that `m_tx_relay == nullptr`. While a check for `m_mempool` wasn't done, a check for `m_tx_relay` is done in the caller of this function, hence a `m_mempool` check is redundant. I am still leaving the check here, while modifying it so that the debug builds fail, let me know if it _should_ be removed.

Reply:
> let me know if it should be removed

I think it is fine to keep it with the `Assume` to document exceptions.